File: outputs/2025-07-30_15-22-09-304114/Source.py
# Unit test result: ✅ PASSED
from flask import Flask, jsonify, request
from typing import List, Optional
app = Flask(__name__)

class Usuario:
    id: Optional[int]
    nome: str
    email: str

    def __init__(self, id: Optional[int], nome: str, email: str):
        self.id = id
        self.nome = nome
        self.email = email

    def to_dict(self):
        return {'id': self.id, 'nome': self.nome, 'email': self.email}

class UsuarioRepository:
    armazenamento: dict[int, Usuario]
    sequence: int

    def __init__(self):
        self.armazenamento = {}
        self.sequence = 0
        self.init_data()

    def init_data(self):
        self.save(Usuario(None, 'Alice', 'alice@example.com'))
        self.save(Usuario(None, 'Bruno', 'bruno@example.com'))

    def find_by_id(self, id: int) -> Optional[Usuario]:
        return self.armazenamento.get(id)

    def find_all(self) -> List[Usuario]:
        return list(self.armazenamento.values())

    def save(self, usuario: Usuario) -> Usuario:
        if usuario.id is None:
            self.sequence += 1
            usuario.id = self.sequence
        self.armazenamento[usuario.id] = usuario
        return usuario

    def delete_by_id(self, id: int):
        self.armazenamento.pop(id, None)

class UsuarioService:
    repo: UsuarioRepository

    def __init__(self, repo: UsuarioRepository):
        self.repo = repo

    def obterTodos(self) -> List[Usuario]:
        return self.repo()

    def obterPorId(self, id: int) -> Optional[Usuario]:
        return self.repo()

    def criar(self, usuario: Usuario) -> Usuario:
        return self.repo()

    def atualizar(self, usuario: Usuario) -> Usuario:
        return self.repo()

    def remover(self, id: int):
        self.repo()
usuario_repo = UsuarioRepository()
usuario_service = UsuarioService(usuario_repo)

@app.route('/api/usuarios', methods=['GET'])
def listar_usuarios():
    usuarios = usuario_service.obterTodos()
    return jsonify([u.to_dict() for u in usuarios])

@app.route('/api/usuarios/<int:id>', methods=['GET'])
def obter_usuario(id: int):
    usuario = usuario_service.obterPorId(id)
    if usuario:
        return jsonify(usuario.to_dict())
    else:
        return (jsonify({'message': 'Usuario not found'}), 404)

@app.route('/api/usuarios', methods=['POST'])
def criar_usuario():
    data = request.json
    new_usuario = Usuario(None, data['nome'], data['email'])
    created_usuario = usuario_service.criar(new_usuario)
    return (jsonify(created_usuario.to_dict()), 201)

@app.route('/api/usuarios/<int:id>', methods=['PUT'])
def atualizar_usuario(id: int):
    data = request.json
    usuario_to_update = Usuario(id, data['nome'], data['email'])
    updated_usuario = usuario_service.atualizar(usuario_to_update)
    if updated_usuario:
        return jsonify(updated_usuario.to_dict())
    else:
        return (jsonify({'message': 'Usuario not found for update'}), 404)

@app.route('/api/usuarios/<int:id>', methods=['DELETE'])
def remover_usuario(id: int):
    usuario_service.remover(id)
    return ('', 204)
if __name__ == '__main__':
    app.run(debug=True)

File: uploads/2025-07-30_15-22-09-304114/Source.java
package com.example.demo;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.web.bind.annotation.*;
import org.springframework.stereotype.Service;
import org.springframework.stereotype.Repository;
import org.springframework.beans.factory.annotation.Autowired;
import javax.annotation.PostConstruct;
import java.util.*;

@SpringBootApplication
public class DemoApplication {

    public static void main(String[] args) {
        SpringApplication.run(DemoApplication.class, args);
    }

    /**
     * Entidade simples para representar um Utilizador
     */
    public static class Usuario {
        private Long id;
        private String nome;
        private String email;

        public Usuario() {}

        public Usuario(Long id, String nome, String email) {
            this.id = id;
            this.nome = nome;
            this.email = email;
        }

        public Long getId() {
            return id;
        }

        public void setId(Long id) {
            this.id = id;
        }

        public String getNome() {
            return nome;
        }

        public void setNome(String nome) {
            this.nome = nome;
        }

        public String getEmail() {
            return email;
        }

        public void setEmail(String email) {
            this.email = email;
        }
    }

    /**
     * Controlador REST para gerir Utilizadores
     */
    @RestController
    @RequestMapping("/api/usuarios")
    public static class UsuarioController {

        @Autowired
        private UsuarioService service;

        @GetMapping
        public List<Usuario> listar() {
            return service.obterTodos();
        }

        @GetMapping("/{id}")
        public Usuario obter(@PathVariable Long id) {
            return service.obterPorId(id);
        }

        @PostMapping
        public Usuario criar(@RequestBody Usuario usuario) {
            return service.criar(usuario);
        }

        @PutMapping("/{id}")
        public Usuario atualizar(@PathVariable Long id, @RequestBody Usuario usuario) {
            usuario.setId(id);
            return service.atualizar(usuario);
        }

        @DeleteMapping("/{id}")
        public void remover(@PathVariable Long id) {
            service.remover(id);
        }
    }

    /**
     * Serviço de negócio para Utilizadores
     */
    @Service
    public static class UsuarioService {

        @Autowired
        private UsuarioRepository repo;

        public List<Usuario> obterTodos() {
            return repo.findAll();
        }

        public Usuario obterPorId(Long id) {
            return repo.findById(id).orElse(null);
        }

        public Usuario criar(Usuario usuario) {
            return repo.save(usuario);
        }

        public Usuario atualizar(Usuario usuario) {
            return repo.save(usuario);
        }

        public void remover(Long id) {
            repo.deleteById(id);
        }
    }

    /**
     * Repositório simulado em memória para Utilizadores
     */
    @Repository
    public static class UsuarioRepository {

        private Map<Long, Usuario> armazenamento = new HashMap<>();
        private Long sequence = 0L;

        @PostConstruct
        public void init() {
            // Preencher com alguns dados iniciais
            save(new Usuario(null, "Alice", "alice@example.com"));
            save(new Usuario(null, "Bruno", "bruno@example.com"));
        }

        public Optional<Usuario> findById(Long id) {
            return Optional.ofNullable(armazenamento.get(id));
        }

        public List<Usuario> findAll() {
            return new ArrayList<>(armazenamento.values());
        }

        public Usuario save(Usuario usuario) {
            if (usuario.getId() == null) {
                usuario.setId(++sequence);
            }
            armazenamento.put(usuario.getId(), usuario);
            return usuario;
        }

        public void deleteById(Long id) {
            armazenamento.remove(id);
        }
    }
}
